crates/iota-types/src/full_checkpoint_content.rs

```diff
@@ -57,11 +57,16 @@ impl CheckpointData {
         eventually_removed_object_refs.into_values().collect()
     }
 
-    pub fn input_objects(&self) -> Vec<&Object> {
-        self.transactions
-            .iter()
-            .flat_map(|tx| &tx.input_objects)
-            .collect()
+    /// Returns all objects that are used as input to the transactions in the
+    /// checkpoint, and already exist prior to the checkpoint.
+    pub fn checkpoint_input_objects(&self) -> BTreeMap<ObjectID, &Object> {
+        let mut checkpoint_input_objects = BTreeMap::new();
+        for tx in self.transactions.iter() {
+            for obj in tx.input_objects.iter() {
+                checkpoint_input_objects.entry(obj.id()).or_insert(obj);
+            }
+        }
+        checkpoint_input_objects
```

Comment:
This can now be further simplified

```suggestion
        self.transactions
            .iter()
            .flat_map(|tx| tx.input_objects.iter().map(|obj| (obj.id(), obj)))
            .collect()
```

Reply:
Unfortunately your suggestion is outdated because the former simplification which was done was actually wrong and we reverted it.